machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +64,21 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
-	// Preemptible indicates if created instance is preemptible
+	// GPUs is a list of GPUs to be attached to the VM.
+	// +optional
+	GPUs []GCPGPUConfig `json:"GPUs,omitempty"`
+	// Preemptible indicates if created instance is preemptible.
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
+	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
+	// This is required to be set to TerminateHostMaintenanceType if you want to provision machine with attached GPUs.
+	// +kubebuilder:validation:Enum=MigrateHostMaintenanceType;TerminateHostMaintenanceType;
```

Comment:
```suggestion
	// This is required to be set to Terminate if you want to provision machine with attached GPUs.
	// Otherwise, allowed values are Migrate and Terminate.  
	// +kubebuilder:validation:Enum=Migrate;Terminate;
```